pkg/new_storage/sftp.go

```diff
@@ -22,19 +22,24 @@ type SFTP struct {
 }
 
 func (sftp *SFTP) Connect() error {
-  f_sftp_key, err := ioutil.ReadFile(sftp.Config.Key)
-  if err != nil {
-    return err
-  }
-  sftp_key, err := ssh.ParsePrivateKey(f_sftp_key)
-  if err != nil {
-    return err
-  }
+	auth_methods := []ssh.AuthMethod{}
+
+	if sftp.Config.Key != "" {
+    f_sftp_key, err := ioutil.ReadFile(sftp.Config.Key)
+    if err != nil { return err }
+    sftp_key, err := ssh.ParsePrivateKey(f_sftp_key)
+    if err != nil { return err }
+
+		auth_methods = append(auth_methods, ssh.PublicKeys(sftp_key))
+	}
+
+	if sftp.Config.Password != "" {
+	  auth_methods = append(auth_methods, ssh.Password(sftp.Config.Password))
+	}
```

Comment:
I think we should add 
```
if sftp.Config.Key=="" && sftp.Config.Password=="" {
   return errors.New("please specify sftp.key or sftp.password for authentification")
}
```

i don't think allow users make access to backup without any authorization is a good idea

Reply:
Done